idb/postgres/postgres.go

```diff
@@ -1476,6 +1477,50 @@ func buildTransactionQuery(tf idb.TransactionFilter) (query string, whereArgs []
 	return
 }
 
+// BlockWithTransactions is part of idb.IndexerDB
```

Comment:
The transaction code here can be replaced with a call to `txTransactions` to prevent the 3rd copy of this logic. `Transactions` should probably be refactored to call `txTransactions`. We could make a similar function to use common code for the block request.